runtime/gc_check/CheckEngine.cpp

```diff
@@ -463,11 +465,13 @@ GC_CheckEngine::checkJ9Object(J9JavaVM *javaVM, J9Object* objectPtr, J9MM_Iterat
 #if defined(J9VM_ENV_DATA64)
 	if (OMR_ARE_ANY_BITS_SET(_cycle->getMiscFlags(), J9MODRON_GCCHK_VALID_INDEXABLE_DATA_ADDRESS) && extensions->objectModel.isIndexable(objectPtr) && javaVM->isIndexableDataAddrPresent) {
 		/* Check that the indexable object has the correct data address pointer */
-		if (!extensions->indexableObjectModel.isValidDataAddr((J9IndexableObject*)objectPtr)) {
+		void *dataAddr = extensions->indexableObjectModel.getDataAddrForIndexableObject((J9IndexableObject *)objectPtr);
+		bool isValidDataAddr = (extensions->indexableObjectModel.isVirtualLargeObjectHeapEnabled() && extensions->largeObjectVirtualMemory->getSparseDataPool()->isValidDataPtr(dataAddr));
+		if (!extensions->indexableObjectModel.isValidDataAddr((J9IndexableObject *)objectPtr, dataAddr, isValidDataAddr)) {
```

Comment:
i prefer we don't proceed with this check if isValidDataAddr is already false
and then indexableObjectModel.isValidDataAddr() could be free of any isVirtualLargeObjectHeapEnabled checks